modules/calendar/calendar_embedder.py

```diff
@@ -55,8 +54,13 @@ async def embed_event_pages(
 					page_num=page_num,
 					max_results=results_per_page,
 				)
-				sender = interaction.send if interaction.message is None else interaction.message.edit
-				await sender(embed=embed)
+				sender = (
+					interaction.send
+					if interaction.message is None
+					else interaction.message.edit
+				)
```

Comment:
Not tested, but probably need something like

```py
interaction.send if not interaction.response.is_done() else interaction.edit_original_message
```

interaction.message will always be None unless it was a component interaction

Reply:
Note that is_done() also is true if it was deferred, so there may be cases where this won't work

Reply:
Hmm we may have to refactor that whole logic, or keep track of it some other way

Reply:
Probably should keep track of whether it sent a message with a flag or something. The logic of "send or edit" could also maybe be extracted to a util function.